src/isomorphism.rs

```diff
@@ -175,54 +180,70 @@ impl Vf2State {
 ///
 /// * Luigi P. Cordella, Pasquale Foggia, Carlo Sansone, Mario Vento;
 ///   *A (Sub)Graph Isomorphism Algorithm for Matching Large Graphs*
-pub fn is_isomorphic(dag0: &PyDiGraph, dag1: &PyDiGraph) -> PyResult<bool> {
-    let g0 = &dag0.graph;
-    let g1 = &dag1.graph;
+pub fn is_isomorphic<Ty>(
+    py: Python,
+    g0: &StableGraph<PyObject, PyObject, Ty>,
+    g1: &StableGraph<PyObject, PyObject, Ty>,
+) -> PyResult<bool>
+where
+    Ty: EdgeType,
+{
+    let inner_temp_g0: StableGraph<PyObject, PyObject, Ty>;
+    let inner_temp_g1: StableGraph<PyObject, PyObject, Ty>;
+    let g0_out = if g0.nodes_removed() {
+        inner_temp_g0 = reindex_graph(py, g0);
+        &inner_temp_g0
+    } else {
+        g0
+    };
+    let g1_out = if g1.nodes_removed() {
+        inner_temp_g1 = reindex_graph(py, g1);
+        &inner_temp_g1
+    } else {
+        g1
+    };
+    let g0 = &g0_out;
+    let g1 = &g1_out;
     if g0.node_count() != g1.node_count() || g0.edge_count() != g1.edge_count()
     {
         return Ok(false);
     }
 
-    let mut st = [Vf2State::new(dag0), Vf2State::new(dag1)];
-    let res = try_match(
-        &mut st,
-        dag0,
-        dag1,
-        &mut NoSemanticMatch,
-        &mut NoSemanticMatch,
-    )?;
+    let mut st = [Vf2State::new(g0), Vf2State::new(g1)];
+    let res =
+        try_match(&mut st, g0, g1, &mut NoSemanticMatch, &mut NoSemanticMatch)?;
     Ok(res.unwrap_or(false))
 }
 
-fn reindex_graph(py: Python, dag: &PyDiGraph) -> PyDiGraph {
+fn reindex_graph<Ty>(
+    py: Python,
+    graph: &StableGraph<PyObject, PyObject, Ty>,
+) -> StableGraph<PyObject, PyObject, Ty>
+where
+    Ty: EdgeType,
+{
     // NOTE: this is a hacky workaround to handle non-contiguous node ids in
     // VF2. The code which was forked from petgraph was written assuming the
     // Graph type and not StableGraph so it makes an implicit assumption on
     // node_bound() == node_count() which isn't true with removals on
     // StableGraph. This compacts the node ids as a workaround until VF2State
     // and try_match can be rewitten to handle this (and likely contributed
     // upstream to petgraph too).
-    let mut new_graph = StableDiGraph::<PyObject, PyObject>::new();
+    let mut new_graph = StableGraph::<PyObject, PyObject, Ty>::default();
     let mut id_map: HashMap<NodeIndex, NodeIndex> = HashMap::new();
-    for node_index in dag.graph.node_indices() {
-        let node_data = dag.graph.node_weight(node_index).unwrap();
+    for node_index in graph.node_indices() {
+        let node_data = graph.node_weight(node_index).unwrap();
         let new_index = new_graph.add_node(node_data.clone_ref(py));
         id_map.insert(node_index, new_index);
     }
-    for edge in dag.graph.edge_references() {
+    for edge in graph.edge_references() {
         let edge_w = edge.weight();
         let p_index = id_map.get(&edge.source()).unwrap();
         let c_index = id_map.get(&edge.target()).unwrap();
```

Comment:
A small cleanup here we can do here while we're updating this code is something like:

```suggestion
        let p_index = id_map[&edge.source()];
        let c_index = id_map[&edge.target()];
```